compile failure due to numpy issues in transformations.c

## Expected behavior ##

pip install build/compiles cleanly

## Actual behavior ##
See https://discord.com/channels/807348386012987462/808088023957897258/1381767573611483176 by @ljwoods2 

Curious if I'm missing something, is anyone else getting a Numpy compatibility issue when trying to build MDA from source with pip? Looks like transformations.c relies on the old numpy API (NPY_IN_ARRAY, etc), and I can't build wheel sucessfully and clear errors like this (long list of them from MDAnalysis/lib/src/transformations/transformations.c) unless I change numpy to an earlier version (<1.24) in the build-system "requires" list:

      MDAnalysis/lib/src/transformations/transformations.c:794:57: error: ‘NPY_ALIGNED’ undeclared (first use in this function); did you mean ‘Py_ALIGNED’?
               *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ALIGNED);


Plus, above should throw a deprecation warning / error instead of just failing, right? I see @orbeckst 's issue (https://github.com/MDAnalysis/mdanalysis/issues/2988) about suppressing the deprecated warnings, but looks like it hasn't been fixed yet, so I should still see the warning? 

Apologies if this is something obvious, but I figured Numpy >2.0 is supposed to be able to handle the old API usage? If this is not just me being dumb I will raise an issue



## Code to reproduce the behavior ##
All I did was:

```
git clone git@github.com:MDAnalysis/mdanalysis.git
git checkout develop
conda create -n broken python=3.11 pip
conda activate broken
cd mdanalysis/package
pip install .
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
- Which version of Python (`python -V`)?
- Which operating system?

--- a/package/CHANGELOG
+++ b/package/CHANGELOG
@@ -15,12 +15,14 @@ The rules for this file:
 
 -------------------------------------------------------------------------------
 ??/??/?? IAlibay, orbeckst, BHM-Bob, TRY-ER, Abdulrahman-PROG, pbuslaev,
-         yuxuanzhuang, yuyuan871111, tanishy7777, tulga-rdn, Gareth-elliott
+         yuxuanzhuang, yuyuan871111, tanishy7777, tulga-rdn, Gareth-elliott,
+         hmacdope
 
 
  * 2.10.0
 
 Fixes
+ *
  * Fix incorrect `self.atom` assignment in SingleFrameReaderBase (Issue #5052, PR #5055)
  * Fixes bug in `analysis/hydrogenbonds.py`: `_donors` and `_hydrogens`
    were not updated when the `acceptors_sel` and `hydrogens_sel` were provided

--- a/package/MDAnalysis/lib/src/transformations/transformations.c
+++ b/package/MDAnalysis/lib/src/transformations/transformations.c
@@ -775,7 +775,7 @@ PyConverter_DoubleArray(
     PyObject *object,
     PyObject **address)
 {
-    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
+    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
      if (*address == NULL) return NPY_FAIL;
      return NPY_SUCCEED;
 }
@@ -808,7 +808,7 @@ PyConverter_DoubleArrayOrNone(
     if ((object == NULL) || (object == Py_None)) {
         *address = NULL;
     } else {
-        *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
+        *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
         if (*address == NULL) {
             PyErr_Format(PyExc_ValueError, "can not convert to array");
             return NPY_FAIL;
@@ -823,7 +823,7 @@ PyConverter_DoubleMatrix44(
     PyObject **address)
 {
     PyArrayObject *obj;
-    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
+    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;
@@ -846,7 +846,7 @@ PyConverter_DoubleMatrix44Copy(
 {
     PyArrayObject *obj;
     *address = PyArray_FROM_OTF(object, NPY_DOUBLE,
-                                NPY_ENSURECOPY|NPY_IN_ARRAY);
+                                NPY_ENSURECOPY|NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;
@@ -868,7 +868,7 @@ PyConverter_DoubleVector3(
     PyObject **address)
 {
     PyArrayObject *obj;
-    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
+    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;
@@ -890,7 +890,7 @@ PyConverter_DoubleVector4(
     PyObject **address)
 {
     PyArrayObject *obj;
-    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
+    *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;
@@ -913,7 +913,7 @@ PyConverter_DoubleVector4Copy(
 {
     PyArrayObject *obj;
     *address = PyArray_FROM_OTF(object, NPY_DOUBLE,
-                                NPY_ENSURECOPY|NPY_IN_ARRAY);
+                                NPY_ENSURECOPY|NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;
@@ -938,7 +938,7 @@ PyConverter_DoubleVector3OrNone(
         *address = NULL;
     } else {
         PyArrayObject *obj;
-        *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_IN_ARRAY);
+        *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_IN_ARRAY);
         if (*address == NULL) {
             PyErr_Format(PyExc_ValueError, "can not convert to array");
             return NPY_FAIL;
@@ -3137,7 +3137,7 @@ py_inverse_matrix(
         goto _fail;
 
     matrix = (PyArrayObject *)PyArray_FROM_OTF(object, NPY_DOUBLE,
-                                               NPY_IN_ARRAY);
+                                               NPY_ARRAY_IN_ARRAY);
     if (matrix == NULL) {
         PyErr_Format(PyExc_ValueError, "not an array");
         goto _fail;

--- a/package/MDAnalysis/lib/src/transformations/transformations.c
+++ b/package/MDAnalysis/lib/src/transformations/transformations.c
@@ -791,7 +791,7 @@ PyConverter_AnyDoubleArray(
         Py_INCREF(object);
         return NPY_SUCCEED;
     } else {
-        *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ALIGNED);
+        *address = PyArray_FROM_OTF(object, NPY_DOUBLE, NPY_ARRAY_ALIGNED);
         if (*address == NULL) {
             PyErr_Format(PyExc_ValueError, "can not convert to array");
             return NPY_FAIL;
@@ -846,7 +846,7 @@ PyConverter_DoubleMatrix44Copy(
 {
     PyArrayObject *obj;
     *address = PyArray_FROM_OTF(object, NPY_DOUBLE,
-                                NPY_ENSURECOPY|NPY_ARRAY_IN_ARRAY);
+                                NPY_ARRAY_ENSURECOPY|NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;
@@ -913,7 +913,7 @@ PyConverter_DoubleVector4Copy(
 {
     PyArrayObject *obj;
     *address = PyArray_FROM_OTF(object, NPY_DOUBLE,
-                                NPY_ENSURECOPY|NPY_ARRAY_IN_ARRAY);
+                                NPY_ARRAY_ENSURECOPY|NPY_ARRAY_IN_ARRAY);
     if (*address == NULL) {
         PyErr_Format(PyExc_ValueError, "can not convert to array");
         return NPY_FAIL;

